samza-api/src/main/java/org/apache/samza/table/descriptors/LocalTableDescriptor.java

```diff
@@ -130,20 +125,64 @@ public D withChangelogReplicationFactor(int replicationFactor) {
   }
 
   @Override
-  protected void generateTableSpecConfig(Map<String, String> tableSpecConfig) {
-    super.generateTableSpecConfig(tableSpecConfig);
+  protected void generateConfig(Config jobConfig, Map<String, String> tableConfig) {
+
+    super.generateConfig(jobConfig, tableConfig);
+
+    JavaTableConfig javaTableConfig = new JavaTableConfig(jobConfig);
+
+    String keySerde = javaTableConfig.getKeySerde(tableId);
+    Preconditions.checkNotNull(keySerde, String.format("Key serde for table %s not found", tableId));
+    addStoreConfig(tableConfig, "key.serde", keySerde);
+
+    String valueSerde = javaTableConfig.getValueSerde(tableId);
+    Preconditions.checkNotNull(valueSerde, String.format("Value serde for table %s not found", tableId));
+    addStoreConfig(tableConfig, "msg.serde", valueSerde);
 
-    tableSpecConfig.put(INTERNAL_ENABLE_CHANGELOG, String.valueOf(enableChangelog));
+    if (sideInputs != null && !sideInputs.isEmpty()) {
+      sideInputs.forEach(si -> Preconditions.checkState(isValidSystemStreamName(si), String.format(
+          "Side input stream %s doesn't confirm to pattern %s", si, SYSTEM_STREAM_NAME_PATTERN)));
+      String formattedSideInputs = String.join(",", sideInputs);
+      addStoreConfig(tableConfig, "side.inputs", formattedSideInputs);
+      addStoreConfig(tableConfig, "side.inputs.processor.serialized.instance",
+          SerdeUtils.serialize("Side Inputs Processor", sideInputsProcessor));
+    }
+
+    // Changelog configuration
     if (enableChangelog) {
-      if (changelogStream != null) {
-        tableSpecConfig.put(INTERNAL_CHANGELOG_STREAM, changelogStream);
+      if (StringUtils.isEmpty(changelogStream)) {
+        String jobName = jobConfig.containsKey("job.name")
```

Comment:
Can use JobConfig.getName.get(). Not providing a job.name should be an error, we shouldn't default to "unknown".

Reply:
JobConfig currently resides in core, and it has a bunch of dependencies in core as well, we could change to use it, once it's moved to api as well.